src/integrationTest/java/uk/gov/hmcts/divorce/citizen/event/CitizenLinkApplicant2Test.java

```diff
@@ -133,9 +131,6 @@ public void givenValidAccessCodeWhenCallbackIsInvokedThenAccessCodeIsRemovedAndS
         assertThatJson(actualResponse)
             .when(IGNORING_EXTRA_FIELDS)
             .isEqualTo(json(expectedCcdAboutToStartCallbackSuccessfulResponse()));
-
-        verify(serviceTokenGenerator).generate();
-        verifyNoMoreInteractions(serviceTokenGenerator);
```

Comment:
is there a reason to remove this ?

Reply:
This fails for me sometimes, and always fails on the PR, because we now have scheduled tasks that run at the same time as the test runs.  We can never know whether multiple calls to the ServiceTokenGenerator occur.

Reply:
I have taken a closer look at this problem and came up with a better solution

Reply:
that's much neater nice one @mapingo